queue/init.lua

```diff
@@ -79,10 +99,15 @@ local function wrap_box_cfg()
         orig_cfg = box.cfg
         box.cfg = cfg_wrapper
     elseif type(box.cfg) == 'table' then
-        -- box.cfg after the first box.cfg call
-        local cfg_mt = getmetatable(box.cfg)
-        orig_call = cfg_mt.__call
-        cfg_mt.__call = cfg_call_wrapper
+        if watchers_supported and box.info.ro_reason == 'orphan' then
+            -- Wait for the orphan instance.
+            fiber.new(orphan_waiter)
+        else
```

Comment:
`orphan` seems like a special case, is it could me more general?
```suggestion
        if watchers_supported and box.info.status ~= 'running' then
            -- Wait for the running state and initialize the queue.
            fiber.new(running_waiter)
        else
```